### PR TITLE
Bael 5649

### PR DESCRIPTION
**Pull request by viktorardelean**
Article code

### CODE REVIEW

**Review by bipster — CHANGES_REQUESTED** [review]
@viktorardelean In addition to the inline notes, can you:
* Run the code through the Baeldung formatter?
* See why your tests are failing the automated suite? My guess is that your tests require the Spring context, which requires the *IntegrationSuite suffix
* For the package name, instead of `handlermethodargumentresolver` maybe something more friendly like `jsonargs`

File: spring-web-modules/spring-mvc-basics-5/src/main/java/com/baeldung/handlermethodargumentresolver/UserController.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+@RequestMapping("/user")
+public class UserController {
+
+    @PostMapping("/process/custom")
+    public ResponseEntity process(@JsonArg("firstName") String firstName, @JsonArg("address.city") String city) {
+        // userService.process(firstName, city)
```
[the remark this answers is not in the document]
Comment (viktorardelean) [reply spring-web-modules/spring-mvc-basics-5/src/main/java/com/baeldung/handlermethodargumentresolver/UserController.java]: done

File: spring-web-modules/spring-mvc-basics-5/src/test/java/com/baeldung/handlermethodargumentresolver/UserControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@WebMvcTest(controllers = UserController.class)
+class UserControllerTest {
```
[the remark this answers is not in the document]
Comment (viktorardelean) [reply spring-web-modules/spring-mvc-basics-5/src/test/java/com/baeldung/handlermethodargumentresolver/UserControllerTest.java]: done

**Review by bipster — APPROVED** [review]
Thank you!

**Pull request merged**